solver/src/test/java/org/chocosolver/solver/constraints/binary/SquareTest.java

```diff
@@ -23,6 +23,57 @@
  */
 public class SquareTest {
 
+    @Test(groups = "10s", timeOut = 60000)
+    public void testBigBoundBound() {
+        Model m = new Model();
+        int n = 6;
+        IntVar[] x = m.intVarArray(n, -n, n, true);
+        IntVar[] x2 = m.intVarArray(n, -n*n,n*n, true);
+        for (int i=0;i<x.length;i++) {
+            m.square(x2[i], x[i]).post();
+        }
+        m.allDifferent(x).post();
+        m.allDifferent(x2).post();
+        while (m.getSolver().solve()) ;
+        assertEquals(m.getSolver().getSolutionCount(), 184320);
+        assertEquals(m.getSolver().getFailCount(), 209153);
+        assertEquals(m.getSolver().getNodeCount(), 577792);
```

Comment:
This is too dependent from the default search. I completely see the interest of the tests (showing the strengthening of filtering with enumerated domains), but the tests might become false if we change the default search and I would rather prefer that they stay true in such a case.